api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -48,6 +48,17 @@ default String name() {
    */
   TableScan newScan();
 
+  /**
+   * Create a new {@link BatchScan batch scan} for this table.
+   *
+   * <p>Once a batch scan is created, it can be refined to project columns and filter data.
+   *
+   * @return a batch scan for this table
+   */
+  default BatchScan newBatchScan() {
```

Comment:
I am wondering if `batch` is the most accurate name. `batch` jobs can also do `incremental` scan. To me, `streaming/batch` is more about execution mode.

This scan is contrasting to the `incremental` scans below. Incremental scan is meant for the difference between two snapshots (start, end). This scan is more like a `snapshot` scan. It scans the table using a specific snapshot (as point of view to the table). Just to bootstrap some brainstorming. what about `newSnapshotScan`?

Reply:
This method is also used by aggregate metadata tables like (`all_manifests`), which scan across all snapshots. That's why `newSnapshotScan` may be misleading.

I am coming from the Spark background that uses `batch` in its scan API. However, I am open to other names. Is there a good word with opposite meaning to incremental?

Reply:
We could call it `V2TableScan` but I am not a big fan of using versions in the name.

Reply:
Thoughts, @stevenzwu @rdblue?

Reply:
yeah, `V2TableScan` is not very meaningful. I see why `SnapshotScan` is not appropriate for metadata table scan. I would be fine with `BatchScan` if we can't find better alternative. 

This is a scan on the table or metadata table's state at some point of time. what about `PointScan`? or just `BaseScan`?

Reply:
I think `BatchScan` makes sense, although I do see the point that "batch" isn't really the opposite of "incremental". `TableScan` works the best, but that's already taken. I don't really care for `SnapshotScan` because that's too specific in a way that we don't really need to state (isn't it almost always a snapshot that gets scanned?).

I'm not sure what a better name would be, so I'd go with `BatchScan`.

Reply:
We usually call something `BaseXXX` if it is a default package-private implementation of an interface. I am also not sure how descriptive `PointScan` would be.

I spent the morning trying to come up with alternative names but I had a hard time. I'd say `Scan` or `TableScan` would work best but they are both taken.

Reply:
@stevenzwu, would you be OK with keeping it as `BatchScan` for now to unblock this change?

Reply:
yes. `BatchScan` seems to be the best option so far. I am totally fine.